metrics/metrics.go

```diff
@@ -0,0 +1,35 @@
+package metrics
+
+import (
+	"time"
+)
+
+// Scope is a namespace wrapper around a stats reporter, ensuring that
+// all emitted values have a given prefix or set of tags.
+type Scope interface {
+	// Counter returns the Counter object corresponding to the name.
+	Counter(name string) Counter
+
+	// Tagged returns a new child scope with the given tags and current tags.
+	Tagged(tags map[string]string) Scope
+
+	// SubScope returns a new child scope appending a further name prefix.
+	SubScope(name string) Scope
+
+	// Timer returns the Timer object corresponding to the name.
+	Timer(name string) Timer
```

Comment:
do we need Time for anything yet? If we're not going to have a use for it in any upcomming PRs, I'd say let's hold off adding it for now.